jobs/schedulers.go

```diff
@@ -230,15 +230,28 @@ func (schedulers *Schedulers) scheduleJob(cfg *model.Config, scheduler model.Sch
 	return scheduler.ScheduleJob(cfg, pendingJobs, lastSuccessfulJob)
 }
 
-func (schedulers *Schedulers) handleConfigChange(oldConfig, newConfig *model.Config) {
+func (schedulers *Schedulers) handleConfigChange(_, newConfig *model.Config) {
 	mlog.Debug("Schedulers received config change.")
-	schedulers.configChanged <- newConfig
+	select {
+	case schedulers.configChanged <- newConfig:
+	case <-schedulers.stop:
+	}
 }
 
-func (schedulers *Schedulers) HandleClusterLeaderChange(isLeader bool) {
+func (schedulers *Schedulers) handleClusterLeaderChange(isLeader bool) {
 	select {
 	case schedulers.clusterLeaderChanged <- isLeader:
```

Comment:
Wondering what will happen if we make this blocking?

Reply:
I am thinking that could cause a deadlock if the schedulers were not started yet when such event came since all those calls are called from methods of `JobServer` under a lock.

Reply:
Ah yeah .. we should really sort out these initialization steps and start the scheduler before setting up clustering. @isacikgoz 's app changes are a first step towards that. But we have more to do :) Anyways, out of scope for this PR.